mlir/lib/Dialect/LLVMIR/IR/LLVMAttrs.cpp

```diff
@@ -380,8 +380,20 @@ LogicalResult
 ModuleFlagAttr::verify(function_ref<InFlightDiagnostic()> emitError,
                        LLVM::ModFlagBehavior flagBehavior, StringAttr key,
                        Attribute value) {
-  if (!isa<IntegerAttr, StringAttr>(value))
-    return emitError()
-           << "only integer and string values are currently supported";
-  return success();
+  if (key == "CG Profile") {
+    auto arrayAttr = dyn_cast<ArrayAttr>(value);
+    if ((!arrayAttr) || (!llvm::all_of(arrayAttr, [](Attribute v) {
```

Comment:
```suggestion
    if ((!arrayAttr) || (!llvm::all_of(arrayAttr, [](Attribute attr) {
```
ultra nit: for attributes we usually use attr or similar.